engine/docker/docker.go

```diff
@@ -157,10 +157,31 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	containers := e.containers
 	e.mu.Unlock()
 
+	timeout := time.Second * 30 // similar to k8s default timeout in pods
 	// stop all containers
 	for _, ctrName := range containers {
-		if err := e.client.ContainerKill(ctx, ctrName, "9"); err != nil {
-			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to kill container")
+		if err := e.client.ContainerStop(ctx, ctrName, &timeout); err != nil {
+			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to stop the container")
+		}
+
+		// Before removing the container we want to be sure that it's in a healthy state to be removed.
```

Comment:
I think we should have 2 for loops: one for loop to stop all containers, one for loop to check status and kill those that didn't stop.

That would parallelize stop.

Reply:
The stop should stop the containers and if after 30 seconds the containers are not stopped, docker will kill them for us so i do not think that we need a second for loop